### PR TITLE
OHRM5X-1267: Fix upgrader bugs

### PR DESCRIPTION
**Pull request by SharukaNP**

### CODE REVIEW

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: devTools/core/src/Command/Util/TranslationTestTool.php [hunk 1]
```diff
@@ -146,4 +153,35 @@ private function getTranslationRecord(int $langStringId, int $langId): string
...
+
+        if (is_null($result)) {
+            $insetQuery = $this->createQueryBuilder();
+            $insetQuery->insert('ohrm_i18n_language')
+                ->values(
+                    ['name' => ':name',
+                        'code' => ':code',
+                        'enabled ' => ':enabled',
+                        'added' => ':added'
+                    ]
```
Comment (RajithaKumara) [inline devTools/core/src/Command/Util/TranslationTestTool.php:178]:
```suggestion
                    [
                        'name' => ':name',
                        'code' => ':code',
                        'enabled ' => ':enabled',
                        'added' => ':added'
                    ]
```

File: installer/Migration/V5_0_0/Migration.php [hunk 2]
```diff
@@ -225,6 +233,24 @@ public function up(): void
...
+            ->where('customFields.type = :type')
+            ->setParameter('type', 1);
+        $results = $q->executeQuery()
+            ->fetchAllAssociative();
+        $extrasData = array_column($results, 'extra_data');
+        foreach ($extrasData as $extraData) {
+            $this->createQueryBuilder()
+                ->update('hs_hr_custom_fields ', 'customFields')
+                ->set('customFields.extra_data', ':newExtraData')
+                ->where('customFields.extra_data = :extraData')
```
Comment (RajithaKumara) [inline installer/Migration/V5_0_0/Migration.php:249]: please based on `hs_hr_custom_fields` primary key

File: installer/upgrader/consoleTest.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
+<?php
+/**
```
Comment (RajithaKumara) [inline installer/upgrader/consoleTest.php:2]: hope we are going to remove this once we complete the upgrader

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/upgrader/consoleTest.php (hunk 3 above)
Comment (SharukaNP) [reply installer/upgrader/consoleTest.php]: removed

File: installer/Migration/V5_0_0/Migration.php [hunk 4]
```diff
@@ -225,6 +234,24 @@ public function up(): void
...
+            ->setParameter('type', 1);
+        $results = $q->executeQuery()
+            ->fetchAllAssociative();
+        foreach ($results as $result) {
+            $customField = CustomField::createFromArray($result);
+            $this->createQueryBuilder()
+                ->update('hs_hr_custom_fields ', 'customFields')
+                ->set('customFields.extra_data', ':newExtraData')
+                ->where('customFields.field_num = :fieldNum')
+                ->setParameter('newExtraData', str_replace(', ', ',', $customField->getExtraData()))
```
Comment (RajithaKumara) [inline installer/Migration/V5_0_0/Migration.php:251]:
```suggestion
                ->setParameter('newExtraData', str_replace(', ', ',', $result['extra_data']))
```

File: installer/Migration/V5_0_0/Migration.php [hunk 5]
```diff
@@ -225,6 +234,24 @@ public function up(): void
...
+
+        $q = $this->getConnection()->createQueryBuilder();
+        $q->select('customFields.extra_data', 'customFields.field_num')
+            ->from('hs_hr_custom_fields ', 'customFields')
+            ->where('customFields.type = :type')
+            ->setParameter('type', 1);
+        $results = $q->executeQuery()
+            ->fetchAllAssociative();
+        foreach ($results as $result) {
+            $customField = CustomField::createFromArray($result);
```
Comment (RajithaKumara) [inline installer/Migration/V5_0_0/Migration.php:246]:
```suggestion
```

File: installer/Util/V1/Dto/CustomField.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<?php
```
Comment (RajithaKumara) [inline installer/Util/V1/Dto/CustomField.php:1]: will remove this class

File: src/client/src/orangehrmLeavePlugin/components/LeaveListTable.vue [hunk 7]
```diff
@@ -379,9 +379,16 @@ export default {
     this.http
       .request({method: 'GET', url: 'api/v2/leave/leave-periods'})
       .then(response => {
-        const {data} = response.data;
-        this.filters.fromDate = this.filters.fromDate ?? data[0]?.startDate;
-        this.filters.toDate = this.filters.toDate ?? data[0]?.endDate;
+        const {data, meta} = response.data;
```
Comment (RajithaKumara) [inline src/client/src/orangehrmLeavePlugin/components/LeaveListTable.vue:382]: couldn't get it through props?

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**